### DISCUSSION TITLE
Why is next rebuilding static pages, causing a FUNCTION_INVOCATION_TIMEOUT, and Cache not working

### DISCUSSION THREAD

**Discussion by GavOutdoors** [vercel/next.js#63124] [Help]
### Summary

I've been struggling with a recently appearing issue. Looking at my Git history, I can't fathom what I did to introduce it. It looks very much like a platform issue.
I run Next 14 on Vercel on the Pro plan.

Nearly all pages are statically generated, revalidating after 24 hours. 
The build completes without a problem.
I test the staging deployment and examine the vercel server logs.

I can see Next pre-fetching the pages... but instead of returning the statically built pages, it is trying to rebuild them and times out.  If I go to a page, I get Function Invocation Timeout. 
![image](https://github.com/vercel/next.js/assets/79079903/dd691304-d261-47ed-b670-b377dda18acb)

I added console.time/timeLog/timeEnd to the code and can see in the logs that the code that is being rendered completes in around 500ms - well within the timeout cutoff.

I also question Next's implementation of React Cache.
Because I am rendering on the server with a database, I am not using the optimised fetch API. Instead, following the Next guidance, expensive serverside processing (such as fetching regular data from the database) is wrapped in the React Cache statement). 
I can see from the logging, that the functions are executing each time, and not hitting the cache. (So, Next's suggestion of don't worry about making the same call again as it is cached, is flawed). 
Looking at the server's Data Cache, there are never any hits.
.... is this also indicating to the issue why the static generated pages are being rebuilt???

I still have an older building working production, but I can't see much difference between the two. Nothing has changed in the next.config, for example.

So, to summarise:
-  Build is generating the static pages.
-  When previewing the staging build, Vercel/Next Js attempts to rebuild the static pages, but then times out.
-  The page code completed well within the time-out period though.
-  React Cache on server-only functions is not working.
-  Everything works fine locally.

Has anyone seen this before?  
Any pointers on where to look next?

Thanks!

### Additional information

Here's an example log output from the staging server in Vercel that reported a Gateway Timeout. 
```
loadPlaceType - caravan: 0.117ms
=== Render Place Page - caravan: 1.253ms
Place type layout is loading
Place Type Loading Page
generating meta data for place type caravan
PlaceTypeMetadata-caravan0.6115412927274195: 0.119ms
loadPlaceType - caravan: 0.01ms
=== Render Place Page - caravan: 0.182ms
Place type layout is loading
Place Type Loading Page
generating meta data for place type caravan
PlaceTypeMetadata-caravan0.8848362708422297: 0.031ms
```
As you can see, the page generated and fetched data quickly and completed well before any timeout threshold. 
(Though not sure why NextJS is trying to build the page twice).

### Example

Note that in the logs there are unspecified errors as well, prior to the 504 timeout.
![image](https://github.com/vercel/next.js/assets/79079903/c6bca840-2fd5-4477-af91-8bfd8d40b99b)

**Comment by icyJoseph** [comment]
Could you share more about you are using cache? Some pseudocode would be helpful.

**Reply by GavOutdoors** [reply]
Caching is as per the documentation [here](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#example)

```
import { cache } from 'react'
 
export const getItem = cache(async (id: string) => {
  const item = await db.item.findUnique({ id })
  return item
})
```

**Comment by GavOutdoors** [comment]
Update: Another NPM update saw many updates to Node. These updates stopped the errors in the Next log on Vercel. No more timeouts  :-D
 
I'm still seeing repeated calls to cached functions though.

**Comment by ShuangDa1018** [comment]
same problem any answer?
